src/URI.js

```diff
@@ -1670,63 +1731,67 @@
     return this;
   };
   p.normalizePath = function(build) {
-    if (this._parts.urn) {
-      return this;
-    }
-
-    if (!this._parts.path || this._parts.path === '/') {
-      return this;
-    }
-
-    var _was_relative;
     var _path = this._parts.path;
-    var _leadingParents = '';
-    var _parent, _pos;
 
-    // handle relative paths
-    if (_path.charAt(0) !== '/') {
-      _was_relative = true;
-      _path = '/' + _path;
-    }
+    if (this._parts.urn) {
```

Comment:
can this be structured in a way that avoids the "christmas tree" effect of indentation?